datafusion/sql/src/unparser/expr.rs

```diff
@@ -316,14 +490,81 @@ mod tests {
 
     #[test]
     fn expr_to_sql_ok() -> Result<()> {
-        let tests: Vec<(Expr, &str)> = vec![(
-            Expr::Column(Column {
-                relation: Some(TableReference::partial("a", "b")),
-                name: "c".to_string(),
-            })
-            .gt(lit(4)),
-            r#"a.b.c > 4"#,
-        )];
+        let tests: Vec<(Expr, &str)> = vec![
+            ((col("a") + col("b")).gt(lit(4)), r#"((a + b) > 4)"#),
+            (
+                Expr::Column(Column {
+                    relation: Some(TableReference::partial("a", "b")),
+                    name: "c".to_string(),
+                })
+                .gt(lit(4)),
+                r#"(a.b.c > 4)"#,
+            ),
+            (
+                Expr::Cast(Cast {
+                    expr: Box::new(col("a")),
+                    data_type: DataType::Date64,
+                }),
+                r#"CAST(a AS DATETIME)"#,
+            ),
+            (
+                Expr::Cast(Cast {
+                    expr: Box::new(col("a")),
+                    data_type: DataType::UInt32,
+                }),
+                r#"CAST(a AS INTEGER UNSIGNED)"#,
+            ),
+            (
+                Expr::Literal(ScalarValue::Date64(Some(0))),
```

Comment:
Since it is not possible to directly represent a Date scalar in a SQL string, these non round trip tests are still needed.